addons/account/tests/test_account_move_in_invoice.py

```diff
@@ -1996,3 +1996,15 @@ def test_in_invoice_reverse_caba(self):
                 'credit': value['debit'],
             })
         self.assertRecordValues(reversed_caba_move.line_ids, expected_values)
+
+    def test_in_invoice_line_tax_line_delete(self):
+        with Form(self.invoice) as invoice_form:
+            lines_count = len(invoice_form.line_ids)
+            with invoice_form.line_ids.edit(0) as line_form:
+                tax = line_form.tax_line_id
+            invoice_form.line_ids.remove(0)
+            # check that the tax line is not deleted/recreated
+            self.assertEqual(len(invoice_form.line_ids), lines_count)
+
+        # Assert the tax line is recreated for the tax
+        self.assertTrue(tax in self.invoice.line_ids.mapped('tax_line_id'))
```

Comment:
Better error message when it fails, no need for `mapped`
```suggestion
        self.assertIn(tax, self.invoice.line_ids.tax_line_id)
```